app/components/hiring_staff/vacancy_form_page_heading_component.rb

```diff
@@ -0,0 +1,52 @@
+class HiringStaff::VacancyFormPageHeadingComponent < ViewComponent::Base
+  delegate :current_organisation, to: :helpers
+  delegate :current_step, to: :helpers
+  delegate :total_steps, to: :helpers
+
+  def initialize(vacancy, session_vacancy_attributes)
+    unless session_vacancy_attributes.nil?
+      @session_job_location = session_vacancy_attributes["job_location"]
+      @session_readable_job_location = session_vacancy_attributes["readable_job_location"]
+    end
+    @vacancy = vacancy
+  end
+
+  def heading
+    vacancy.present? ? page_title : page_title_no_vacancy
+  end
+
+  def show_current_step?
+    %w[copy edit edit_published].exclude?(vacancy&.state)
+  end
+
+private
+
+  attr_reader :session_job_location, :session_readable_job_location, :vacancy
```

Comment:
stuff like the placement of this line would be nice to be consistent across all components

Reply:
What's the alternative placement?

I want to restrict the interface of the component as far as possible by not keeping attr_readers private where possible.

Reply:
That's cool, just would be nice (whatever we do) to do the same across the different components if possible